### PR TITLE
[macsec]: add SAI_MACSEC_ATTR_SUPPORTED_CIPHER_SUITE_LIST

### PR DESCRIPTION
**Pull request by dipankar-ba**
…nsistent with the PR1156 style change in SAI_MACSEC_SC_ATTR

Signed-off-by: dipankar-ba <dipankar@arista.com>

### CODE REVIEW

File: inc/saimacsec.h [hunk 1]
```diff
@@ -118,36 +118,12 @@ typedef enum _sai_macsec_attr_t
...
-    SAI_MACSEC_ATTR_GCM_AES128_SUPPORTED,
-
-    /**
-     * @brief Indicates if GCM-AES256 cipher-suite is supported.
-     *
-     * @type bool
+     * @type sai_s32_list_t sai_macsec_cipher_suite_t
      * @flags READ_ONLY
      */
-    SAI_MACSEC_ATTR_GCM_AES256_SUPPORTED,
```
Comment (lguohan) [inline inc/saimacsec.h:150]: I think you can mark these attribute deprecated, not removing the immediately. we can remove it in 3~6 months.

File: inc/saimacsec.h [hunk 2]
```diff
@@ -118,31 +118,43 @@ typedef enum _sai_macsec_attr_t
...
-     * @brief Indicates if 32-bit Packer Number (PN) is supported.
+     * @brief List of supported cipher-suites
+     *
+     * @type sai_s32_list_t sai_macsec_cipher_suite_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_SUPPORTED_CIPHER_SUITE_LIST,
+
+    /**
+     * @brief Indicates if 32-bit Packer Number (PN) is supported.  This is To be deprecated
```
Comment (kcudnik) [inline inc/saimacsec.h:129]: there is a flag "@depreacated true" which can be added to flags

File: meta/parse.pl [hunk 3]
```diff
@@ -265,12 +265,13 @@ sub ProcessTagIsRecourceType
 
 sub ProcessTagDeprecated
 {
-    # just return true if defined
+    # just return true if not defined
```
Comment (kcudnik) [inline meta/parse.pl:268]: previous statement was correct

File: meta/parse.pl [hunk 4]
```diff
@@ -265,12 +265,13 @@ sub ProcessTagIsRecourceType
...
 
     my ($type, $value, $val) = @_;
 
-    LogError "deprecated tag should not have value '$val'" if not $val =~ /^$/i;
+    return $val if $val =~ /^(true|false)$/i;
 
+    LogError "deprecated tag value '$val', expected true/false";
```
Comment (kcudnik) [inline meta/parse.pl:274]: this is similar to isresourcetype, return "undef" maybe here?

File: meta/parse.pl (hunk 3 above)
Comment (kcudnik) [reply meta/parse.pl:268]: actually this comment could be removed

**Review by kcudnik — CHANGES_REQUESTED** [review]

File: inc/saimacsec.h (hunk 1 above)
Comment (dipankar-ba) [reply inc/saimacsec.h:150]: ok

File: inc/saimacsec.h (hunk 2 above)
Comment (dipankar-ba) [reply inc/saimacsec.h]: Done

File: meta/parse.pl (hunk 3 above)
Comment (dipankar-ba) [reply meta/parse.pl]: Done

File: meta/parse.pl (hunk 4 above)
Comment (dipankar-ba) [reply meta/parse.pl:272]: Changed to undef

**Review by kcudnik — APPROVED** [review]

**Review by prafull-brcm — COMMENTED** [review]
Changes looks good to me.

**Review by lguohan — APPROVED** [review]

**Pull request merged**